acme/acme/standalone.py

```diff
@@ -0,0 +1,230 @@
+"""Support for standalone client challenge solvers. """
+import argparse
+import collections
+import functools
+import logging
+import os
+import socket
+import sys
+
+import six
+from six.moves import BaseHTTPServer  # pylint: disable=import-error
+from six.moves import http_client  # pylint: disable=import-error
+from six.moves import socketserver  # pylint: disable=import-error
+
+import OpenSSL
+
+from acme import challenges
+from acme import crypto_util
+
+
+logger = logging.getLogger(__name__)
+
+# six.moves.* | pylint: disable=no-member,attribute-defined-outside-init
+# pylint: disable=too-few-public-methods,no-init
+
+
+class TLSServer(socketserver.TCPServer):
+    """Generic TLS Server."""
+
+    def __init__(self, *args, **kwargs):
+        self.certs = kwargs.pop("certs", {})
+        self.method = kwargs.pop(
+            # pylint: disable=protected-access
+            "method", crypto_util._DEFAULT_DVSNI_SSL_METHOD)
+        self.allow_reuse_address = kwargs.pop("allow_reuse_address", True)
+        socketserver.TCPServer.__init__(self, *args, **kwargs)
+
+    def _wrap_sock(self):
+        self.socket = crypto_util.SSLSocket(
+            self.socket, certs=self.certs, method=self.method)
+
+    def server_bind(self):  # pylint: disable=missing-docstring
+        self._wrap_sock()
+        return socketserver.TCPServer.server_bind(self)
+
+
+class HTTPSServer(TLSServer, BaseHTTPServer.HTTPServer):
+    """HTTPS Server."""
+
+    def server_bind(self):
+        self._wrap_sock()
+        BaseHTTPServer.HTTPServer.server_bind(self)
+
+
+class ACMEServerMixin:  # pylint: disable=old-style-class,no-init
+    """ACME server common settings mixin.
+
+    .. warning::
+       Subclasses have to init ``_stopped = False`` (it's not done here,
+       because of old-style classes madness).
+
+    """
```

Comment:
Personally, I'd add an `__init__` function that derived classes can call instead of relying on them to setup `_stopped`.